### PR TITLE
Change log and publishing updates

### PR DESCRIPTION
**Pull request by jeffb-sfdc**
Change log and publishing updates

### What does this PR do?
Updates to publishing.ms and changelog.md

### What issues does this PR fix or reference?
@W-12557973@

### What does this PR do?
* Changed the URL in CHANGELOG.md to a relative path.
(the link now works in branches)
* Made minor updates to the publishing doc (contributing/publishing.md), mostly comprising of minor formatting changes and some updated steps in the "Merging the Release Branch into Main" and "Publishing Main" sections.

### CODE REVIEW

File: contributing/publishing.md [hunk 1]
```diff
@@ -22,38 +22,44 @@ For more information about publishing take a look at:
...
 
-The merge into main will trigger a run of the 'Test, Build, and Release' GHA workflow that will:
+The merge into `main` will trigger a run of the 'Test, Build, and Release' GHA workflow (https://github.com/forcedotcom/salesforcedx-vscode/actions/workflows/testBuildAndRelease.yml) that will:
 - run the tests
 - build vsix files
 - send a slack notification that a release workflow has been initiated
-- create a tag and release in Github
+- create a tag and release in GitHub
+
+In the 'Test, Build, and Release' workflow:
```
Comment (randi274) [inline contributing/publishing.md:53]: Should this come after line 62, for the instructions on how to "...approve the publish job"?

**Comment by gbockus-sf** [comment]
You'll need to associated this PR with a WI to get the validation to pass.  Probably could use W-12552851 (ticket for gha publish polish)

File: contributing/publishing.md (hunk 1 above)
Comment (gbockus-sf) [reply contributing/publishing.md:53]: heh...we actually have it both the Test, Build <pause>, and Release job and the publish job
We probably don't need the pause in the build job b/c I can't envision a time when you'd want to abort after the tests/build happens but pre release/tag/merge.  I left it in partially due to just being overly careful with the process working as expected. 

At this point I could just take it out.  thoughts?

**Review by gbockus-sf — APPROVED** [review]
LGTM.  Thanks for updating the doc!

**Comment by jeffb-sfdc** [comment]
> 

@gbockus-sf I created a work item and referenced it in this ticket, but the `pr-validation` job is still failing.  The error it gives is:
"GUS WIs should be wrapped in @s, ex: @W-12345678@ "

...and this is in this work item.

**Comment by gbockus-sf** [comment]
> > 
> 
> @gbockus-sf I created a work item and referenced it in this ticket, but the `pr-validation` job is still failing. The error it gives is: _"GUS WIs should be wrapped in @s, ex: @W-12345678@..."_
> 
> ...and this is in this work item. Any idea how to get past this?

Maybe try closing and reopening the PR?  If that doesn't get past it I know creating a new PR will do it (or I just just click the red button since this is just a doc update).

**Comment by gbockus-sf** [comment]
@jeffb-sfdc note I closed and reopened with my previous comment. Let's see if it passes now.

**Pull request merged**